base/server/upgrade/10.9.0/05-FixInstanceNSSDbFolderOwnership.py

```diff
@@ -0,0 +1,30 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+from __future__ import absolute_import
+import os
+import pki.server.upgrade
+
+
+class FixInstanceNSSDbFolderOwnership(
+        pki.server.upgrade.PKIServerUpgradeScriptlet):
+
+    def __init__(self):
+        super(FixInstanceNSSDbFolderOwnership, self).__init__()
+        self.message = 'Fix instance nssdb directory ownership'
+
+    def upgrade_instance(self, instance):
+        directory = instance.nssdb_dir
+        if os.path.exists(directory):
+            pki.util.chown(directory, instance.uid, instance.gid)
```

Comment:
Question for @edewata:
The patch seems to fix `pkispawn` scriptlet. In existing instances, will this line have any effect? IIUC, this will still set the gid to whatever gid was used originally during pkispawn.

Reply:
The upgrade now runs as `pkiuser` during server startup, so if the files are incorrectly owned by another user this operation might not work.

I'm trying to add a CLI that will reset the permission of all files in the instance. That CLI can be executed during RPM upgrade (in the RPM spec) as root, so it should be able to fix all permission issues.

So for this PR I'd suggest to remove the upgrade script, but keep the change in `security_databases.py`.

Reply:
Yes, the upgrade procedure is run under a non-privileged user(`pkiuser`). According to \[0\]:

> Only a privileged process (Linux: one with the CAP_CHOWN capability)
may change the owner of a file.  The owner of a file may change the
group of the file to any group of which that owner is a member.  A
privileged process (Linux: with CAP_CHOWN) may change the group
arbitrarily.

So, my upgrade script works only if the process has `CAP_CHOWN` or EUID is the same as the owner of that files. I'm wrong there. A dedicated script/executable having `CAP_CHOWN` will be the best option, I think.

\[0\]: https://man7.org/linux/man-pages/man2/chown.2.html